src/test/java/com/gildedrose/ItemsControllerTest.java

```diff
@@ -0,0 +1,58 @@
+package com.gildedrose;
+
+import com.gildedrose.goblins_grotto.Item;
+import com.gildedrose.items.Ageable;
+import com.gildedrose.items.DefaultItem;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import waiter.Request;
+import waiter.Response;
+
+import java.util.ArrayList;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class ItemsControllerTest {
+
+    ArrayList<Ageable> mockedDB;
+    ItemsController itemsController;
+
+    @BeforeEach
+    void setUp() {
+        mockedDB = new ArrayList<>();
+        itemsController = new ItemsController(new ItemsDataSourceMock<>(mockedDB));
+        mockedDB.add(new DefaultItem(new Item("foo", 3, 14)));
+
+    }
+
+    @Test
+    void returnsOkResponse_WhenSingleItemIsRetrieved() {
+        Request request = new Request("/inventory", Request.Method.GET.asString, "foo", "foo", "foo");
+        request.setParameter("1");
+
+        Response response = itemsController.okSingleItemHandler.apply(request);
+
+        assertEquals(Response.Status.OK, response.getStatus());
```

Comment:
How do we know the item requested is returned and is the correct one?

Reply:
That is a good point, now checks if the body has the item name in it! This is addressed in [this commit here](https://github.com/jramnani/GildedRoseJavaLLC/pull/24/commits/1983bd7ab5c6f63685323b51339047b4abb90ede)!